qiskit/circuit/quantumcircuit.py

```diff
@@ -1508,9 +1510,11 @@ def decompose(
         from qiskit.converters.circuit_to_dag import circuit_to_dag
         from qiskit.converters.dag_to_circuit import dag_to_circuit
 
-        pass_ = Decompose(gates_to_decompose=gates_to_decompose)
-        decomposed_dag = pass_.run(circuit_to_dag(self))
-        return dag_to_circuit(decomposed_dag)
+        for _ in range(reps):
+            pass_ = Decompose(gates_to_decompose=gates_to_decompose)
+            decomposed_dag = pass_.run(circuit_to_dag(self))
+            self = dag_to_circuit(decomposed_dag)
```

Comment:
Does this overwrite the circuit inplace since you re-assign the `self` variable? It might be safer to do something like
```python
pass_ = Decompose(...)
decomposed_dag = pass_.run(circuit_to_dag(self))
for _ in range(reps - 1):
    decomposed_dag = pass_.run(decomposed_dag)

return dag_to_circuit(decomposed_dag)
```

Reply:
thank you